src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java

```diff
@@ -131,7 +132,7 @@ void updateAdminPassword() {
             final PasswordValidator passwordValidator = PasswordValidator.of(
                 Settings.builder()
                     .put(SECURITY_RESTAPI_PASSWORD_VALIDATION_REGEX, "(?=.*[A-Z])(?=.*[^a-zA-Z\\\\d])(?=.*[0-9])(?=.*[a-z]).{8,}")
-                    .put(SECURITY_RESTAPI_PASSWORD_MIN_LENGTH, 8)
+                    .put(SECURITY_RESTAPI_PASSWORD_MIN_LENGTH, DEFAULT_PASSWORD_MIN_LENGTH)
```

Comment:
Taking a closer look at this, the [docs](https://opensearch.org/docs/latest/security/configuration/yaml/#password-settings) say this is configurable via `plugins.security.restapi.password_min_length` but is it actually configurable if it's being set to `8` all the time? I might be totally missing something though.

Reply:
It would be strange to have different requirements in this tool from the OpenSearch instance that is being setup.

@DarshitChanpura Can you create an issue around this?  The settings configuration should be read from a singular OpenSearch settings, not a separate setting configuration or hardcoded.

Reply:
+1, not sure but it seems like the regex also has some concept of length, maybe we can combine them?

Reply:
`plugins.security.restapi.password_min_length` was introduced in https://github.com/opensearch-project/security/pull/2557 for password validation. 

The password length is not configurable for admin user since the demo script doesn't read settings from opensearch.yml as it is executed before reading the yml file. Remember, admin password is changeable, and the password validation introduced in #2557 should come into play when verifying new password. For changes related to demo script, specifying that the password has to be 8 characters or longer should be enough. 

Let me know if I missed something here.

Reply:
>The password length is not configurable for admin user since the demo script doesn't read settings from opensearch.yml as it is executed before reading the yml file. Remember, admin password is changeable, and the password validation introduced in #2557 should come into play when verifying new password. For changes related to demo script, specifying that the password has to be 8 characters or longer should be enough.

I'm not saying we need to implement it, but we should document that the password requirements for this tool that ships with OpenSearch does not use the same settings.  If someone was to implement this functionality I would approve merging it, what do you think @DarshitChanpura ?

Reply:
done. https://github.com/opensearch-project/security/issues/4081